Makefile

```diff
@@ -685,7 +685,8 @@ netconfig_deploy: input netconfig_deploy_prep ## installs the service instance u
 .PHONY: netconfig_deploy_cleanup
 netconfig_deploy_cleanup: ## cleans up the service instance, Does not affect the operator.
 	$(eval $(call vars,$@,infra))
-	oc delete --all netconfig --ignore-not-found=true || true
+	oc kustomize ${DEPLOY_DIR} | oc delete --ignore-not-found=true -f -
+	rm -Rf ${OPERATOR_BASE_DIR}/infra-operator ${DEPLOY_DIR}
```

Comment:
We have an issue with the overloading of the `infra-operator` dir.  Both `NetConfig` and `Memcached` use the same directory path for their CRs and thus would delete each other's content in that dir during `make (memcached|netconfig)_deploy_cleanup` [1].  They wouldn't be deleting the corresponding resource from the cluster, but when one of the targets removes this dir during `make <whatever>_deploy_cleanup` it would then prevent the other's `make <whatever>_deploy_cleanup` from properly removing the resource from the cluster if it was executed (because there would be no dir nor CR which to feed to `oc delete`).  I'm not sure at the moment what the best way to fix this is.

[1] https://github.com/openstack-k8s-operators/install_yamls/blob/b9f2345ffbdffc37ce0951b7f16d7ad036016630/Makefile#L708-L712

Reply:
Actually, the overloaded directory _would_ remove both the `Memcached` and `NetConfig` resources from the cluster whenever you execute `*_deploy_cleanup`, because the command is:

```
oc kustomize ${DEPLOY_DIR} | oc delete --ignore-not-found=true -f -
```

Any CRs in the `DEPLOY_DIR` would be deleted, and both `NetConfig` and `Memcached` could be in there.

Reply:
Maybe it might help to append `/memcached` and `/netconfig` to `DEPLOY_DIR` in the respective targets?  That might prevent them from deleting each other's resources.

Reply:
Are you deploying memcached separately with `memcached_deploy`? Isn't it deployed with the openstack CR[1]? We only copy one CR to the $DEPLOY_DIR with `netconfig_deploy`[2], so that should only be cleaned up?

[1]https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/samples/core_v1beta1_openstackcontrolplane_galera.yaml#L43
[2] https://github.com/openstack-k8s-operators/install_yamls/blob/main/Makefile#L676

Reply:
We can deal with this later on, as the concern here is not the normal path most people/CI use